Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/springcloud/SpringCloudAppNode.java

```diff
@@ -0,0 +1,159 @@
+    private static final String ACTION_DELETE = "Delete";
+    private static final String ACTION_RESTART = "Restart";
+    private static final String ACTION_OPEN_IN_BROWSER = "Open In Browser";
+
+    protected static final String ACTION_SHOW_PROPERTY = "Show Properties";
+    private final SpringCloudAppNodePresenter springCloudAppNodePresenter;
+    private DeploymentResource deploy;
+
+    public SpringCloudAppNode(AppResourceInner app, DeploymentResource deploy, SpringCloudNode parent) {
+        super(app.id(), String.format("%s - %s", app.name(), getStatus(deploy)), parent,
+                String.format("azure-springcloud-app-%s.png", getStatusIcon(deploy)));
+        this.app = app;
+        this.deploy = deploy;
+        this.clusterName = parent.getServiceName();
+        springCloudAppNodePresenter = new SpringCloudAppNodePresenter();
+        springCloudAppNodePresenter.onAttachView(this);
+
+        loadActions();
+    }
+
+    public String getClusterName() {
+        return clusterName;
+    }
+
+    public String getAppName() {
+        return app.name();
+    }
+
+    public String getAppId() {
+        return app.id();
+    }
+
+    public String getResourceGroup() {
+        return getResourceGroup(this.app.id());
+    }
+
+    public static String getResourceGroup(String serviceId) {
+        final String[] attributes = serviceId.split("/");
+        return attributes[ArrayUtils.indexOf(attributes, "resourceGroups") + 1];
+    }
+
+    @Override
+    protected void loadActions() {
+        addAction(ACTION_START, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, START_SPRING_CLOUD_APP,
+                createBackgroundActionListener("Starting", () -> startSpringCloudApp())));
+        addAction(ACTION_STOP, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, STOP_SPRING_CLOUD_APP,
+                createBackgroundActionListener("Stopping", () -> stopSpringCloudApp())));
+
+        addAction(ACTION_RESTART, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, RESTART_SPRING_CLOUD_APP,
+                createBackgroundActionListener("Restarting", () -> restartSpringCloudApp())));
+        addAction(ACTION_DELETE, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, DELETE_SPRING_CLOUD_APP,
+                createBackgroundActionListener("Deleting", () -> deleteSpringCloudApp())));
+        addAction(ACTION_OPEN_IN_BROWSER, new WrappedTelemetryNodeActionListener(SPRING_CLOUD, OPEN_IN_BROWSER_SPRING_CLOUD_APP,
+                new NodeActionListener() {
+                    @Override
+                    protected void actionPerformed(NodeActionEvent e) {
+                        if (StringUtils.isNotBlank(app.properties().url())) {
+                            DefaultLoader.getUIHelper().openInBrowser(app.properties().url());
+                        } else {
+                            DefaultLoader.getUIHelper().showInfo(SpringCloudAppNode.this, "Public url is not available for app: " + app.name());
```

Comment:
do we need to show another info that public access is not enabled  when `!app.properties().publicProperty()`?